src/plugin.rs

```diff
@@ -108,21 +138,39 @@ impl MigrationSource<'static> for MigrationList {
 }
 
 #[command]
-async fn load(
+async fn load<R: Runtime>(
+    app: AppHandle<R>,
     db_instances: State<'_, DbInstances>,
     migrations: State<'_, Migrations>,
     db: String,
-) -> Result<()> {
-    if !Db::database_exists(&db).await.unwrap_or(false) {
-        Db::create_database(&db).await?;
+) -> Result<String> {
+    // fully qualified DB connection string
+    let fqdb = match cfg!(feature = "sqlite") {
+        true => path_mapper(app_path(&app), db.clone()),
+        false => db.clone(),
+    };
+
+    #[cfg(feature = "sqlite")]
+    create_dir_all(app_path(&app)).expect("Problem creating App directory!");
+
+    let db_exists = Db::database_exists(&fqdb).await.unwrap_or(false);
+
+    if !db_exists {
+        Db::create_database(&fqdb).await?;
     }
-    let pool = Pool::connect(&db).await?;
+
+    let pool = match Pool::connect(&fqdb).await {
+        Ok(p) => p,
+        Err(e) => panic!("Problem establishing pooled DB connection: {:?}", e),
+    };
```

Comment:
I couldn't figure another way to augment the error message but I think reverting to the way it was is better. I added this primarily because I thought this might be the area that had been failing earlier.